drainer/collector_test.go

```diff
@@ -0,0 +1,374 @@
+}
+
+func (s *newCollectorSuite) TestFailedToCreatePDCli(c *C) {
+	origNew := newStore
+	defer func() { newStore = origNew }()
+	newStore = func(path string) (kv.Storage, error) {
+		return dummyStore{}, nil
+	}
+
+	origNewCli := newClient
+	defer func() { newClient = origNewCli }()
+	newClient = func(endpoints []string, dialTimeout time.Duration, root string, security *tls.Config) (*etcd.Client, error) {
+		return nil, errors.New("etcd")
+	}
+
+	cfg := Config{EtcdURLs: "http://localhost:9090"}
+	_, err := NewCollector(&cfg, 0, nil, dummyCheckpoint{})
+	c.Assert(err, ErrorMatches, "etcd")
+}
+
+func (s *newCollectorSuite) TestReturnCollector(c *C) {
+	origNew := newStore
+	defer func() { newStore = origNew }()
+	newStore = func(path string) (kv.Storage, error) {
+		return dummyStore{}, nil
+	}
+
+	origNewCli := newClient
+	defer func() { newClient = origNewCli }()
+	newClient = func(endpoints []string, dialTimeout time.Duration, root string, security *tls.Config) (*etcd.Client, error) {
+		return &etcd.Client{}, nil
+	}
+
+	cfg := Config{EtcdURLs: "http://localhost:9090"}
+	collector, err := NewCollector(&cfg, 1984, nil, dummyCheckpoint{})
+	c.Assert(err, IsNil)
+	c.Assert(collector.clusterID, Equals, uint64(1984))
+}
+
+type syncBinlogSuite struct{}
+
+var _ = Suite(&syncBinlogSuite{})
+
+func (s *syncBinlogSuite) TestShouldAddToSyncer(c *C) {
+	syncer := Syncer{
+		input: make(chan *binlogItem, 1),
+	}
+	col := Collector{syncer: &syncer}
+
+	item := binlogItem{
+		binlog: &pb.Binlog{DdlJobId: 0, CommitTs: 2311945},
+	}
+
+	err := col.syncBinlog(&item)
+
+	c.Assert(err, IsNil)
+	c.Assert(len(syncer.input), Equals, 1)
+	get := <-syncer.input
+	c.Assert(get.binlog.CommitTs, Equals, item.binlog.CommitTs)
+}
+
+func (s *syncBinlogSuite) TestGetDDLFailed(c *C) {
+	origDDLGetter := fDDLJobGetter
+	fDDLJobGetter = func(tiStore kv.Storage, id int64) (*model.Job, error) {
+		return nil, errors.New("getDDLJob")
+	}
+	defer func() { fDDLJobGetter = origDDLGetter }()
+
+	origRetryWait := getDDLJobRetryWait
+	getDDLJobRetryWait = time.Millisecond
+	defer func() { getDDLJobRetryWait = origRetryWait }()
+
+	col := Collector{}
+
+	item := binlogItem{
+		binlog: &pb.Binlog{DdlJobId: 1024, CommitTs: 1305432},
+	}
+
+	err := col.syncBinlog(&item)
+	c.Assert(err, ErrorMatches, "getDDLJob")
+}
+
+func (s *syncBinlogSuite) TestShouldSetJob(c *C) {
+	job := model.Job{
+		ID:    12345,
+		State: model.JobStateSynced,
+	}
+
+	origDDLGetter := fDDLJobGetter
+	fDDLJobGetter = func(tiStore kv.Storage, id int64) (*model.Job, error) {
+		return &job, nil
+	}
+	defer func() { fDDLJobGetter = origDDLGetter }()
+
+	syncer := Syncer{
+		input: make(chan *binlogItem, 1),
+	}
+	col := Collector{syncer: &syncer}
+
+	item := binlogItem{
+		binlog: &pb.Binlog{DdlJobId: 1024, CommitTs: 1305432},
+	}
+
+	err := col.syncBinlog(&item)
+	c.Assert(err, IsNil)
+
+	c.Assert(item.job.ID, DeepEquals, job.ID)
+	c.Assert(item.job.State, DeepEquals, job.State)
```

Comment:
Either you mean `c.Assert(item.job, DeepEquals, job)` or you mean to just use `Equals` here.

Reply:
Guess I tried to use `c.Assert(item.job, DeepEquals, job)` and then found out it was a huge struct and then gave up.